fix(sync): fix API synchronization when environment has no HrID

marcambier opened the pull request:
**Issue**

https://github.com/gravitee-io/issues/issues/7053

**Description**

fix(sync): fix API synchronization when environment has no HrID

An environment may have no HrId.
For example, running APIM on JDBC, the environment_hrids table is empty.

When enhancing API definition with org and env HR ids, we take the first HR id of the list, whithout checking if its empty.
In case of an empty list, it leads to a NoSuchElementException.

<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-qghbbhgckf.chromatic.com)
<!-- Storybook placeholder end -->
<!-- UI placeholder -->
🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/fix-7053/index.html)
_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.
<!-- UI placeholder end -->

Commit:
chore: improve ApiSynchronizer exceptions logging

Commit:
fix(sync): fix API synchronization when environment has no HrID

An environment may have no HrId.
For example, running APIM on JDBC, the environment_hrids table is empty.

When enhancing API definition with org and env HR ids, we take the first HR id of the list, whithout checking if its empty.
In case of an empty list, it leads to a NoSuchElementException.

jhaeyaert reviewed (approved):
This improve the resilience of the sync process 👍 
I'm still a bit confused by the fact the hrid is not defined. In a normal situation we must always have an hrid (not null, not empty). I think there is a kind of issue on the jdbc repository side.

Commit:
fix(jdbc): fix JDBC default environment initialization

With a liquibase script, add a default HRID to default environment if he doesn't have one yet.
Default environment should always have an HRID.

JDBC environment initialization from liquibase will behave the same way that DefaultEnvironmentUpgrader, creating the default environment with his HR ID.
